controllers/events_controller.js

```diff
@@ -167,6 +205,78 @@ module.exports.updateParticipant = async (req, res) => {
 	sendSuccess(res, participant);
 };
 
+module.exports.forgotPassword = async (req, res) => {
+	let { email } = req.body;
+	email = String(email).trim().toLowerCase();
+	let [participant, resetToken] = await Promise.all([
+		Participant.findOne({ email }),
+		ResetToken.findOne({ email })
+	]);
+	if (!participant) {
+		return sendError(res, "No Profile Found", BAD_REQUEST);
+	}
+	let promises = [];
+	if (resetToken) {
+		promises.push(resetToken.delete());
+	}
+
+	let newResetToken = new ResetToken({
+		email: participant.email,
+		id: participant._id,
+		token: `${generateHash(3)}${Date.now()}${generateHash(3)}`,
+		expires: Date.now() + 3600000
+	});
+
+	promises.push(newResetToken.save());
+	let args = {
+		jobName: "sendSystemEmailJob",
+		time: Date.now(),
+		params: {
+			email,
+			name: participant.name,
+			link: `${FRONTEND_URL}/reset/${participant._id}/${newResetToken.token}`,
+			mailType: "pwd-reset-link"
+		}
+	};
+	kue.scheduleJob(args);
+	await Promise.all(promises);
+	return sendSuccess(res, null);
+};
+
+module.exports.resetPassword = async (req, res) => {
+	let { token, id, pwd } = req.body;
+
+	let [participant, resetToken] = await Promise.all([
+		Participant.findById(id),
+		ResetToken.findOne({
+			$and: [{ id }, { token }, { expires: { $gte: Date.now() } }]
+		})
+	]);
+	if (!participant || !resetToken) {
+		return sendError(
+			res,
+			"Reset link is not valid or expired.",
+			BAD_REQUEST
+		);
+	}
+
+	participant.password = String(pwd).trim();
```

Comment:
Are we actually saving the plaintext password here? or is it being encrypted through the `pre-save` hook when participant document updates

Reply:
Encrypted through pre-save hook